test/runtests.jl

```diff
@@ -41,4 +40,44 @@ using Test
             @test CachedCalls._extract_kwargs(fargs) == [(:a, 1), (:b, 2), (:c, :c)]
         end
     end
+
+    @testset "@cached_call" begin
+        @testset "f()" begin
+            f() = 2
+            @test f() == @cached_call(f()) == @cached_call f() # test first and second call
+        end
+
+        @testset "f(a, b)" begin
+            f(a, b) = a + b
+            @test f(1, 2) == @cached_call f(1, 2)
+            @test f(1, 2) != @cached_call f(2, 2)
+        end
+
+        @testset "f(;kw=kw)" begin
+            f(;kw=1) = kw
+            call1 = @cached_call f(;kw=1)
+            call2 = @cached_call f(kw=1)
+            kw = 1
+            if VERSION >= v"1.5"
+                call3 = @cached_call f(;kw)
+                @test f(;kw=1) == call1 == call2 == call3
+            else
+                @test f(;kw=1) == call1 == call2 == call3
+            end
+        end
+
+        @testset "f(;kw=kw)" begin
+            f(a; kw=1) = a - kw
+            a = 2.0
+            kw = 1
+            call1 = @cached_call f(a; kw=1)
```

Comment:
what will happen if I call `f(a)`? Will it recompute or load a cache?

Reply:
It will recompute, I have planned tests for this, just haven't gottten round to them
https://github.com/mzgubic/CachedCalls.jl/issues/6